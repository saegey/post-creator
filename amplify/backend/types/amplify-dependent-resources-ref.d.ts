export type AmplifyDependentResourcesAttributes = {
  "api": {
    "api12660653": {
      "ApiId": "string",
      "ApiName": "string",
      "RootUrl": "string"
    },
    "nextjsblog": {
      "GraphQLAPIEndpointOutput": "string",
      "GraphQLAPIIdOutput": "string",
      "GraphQLAPIKeyOutput": "string"
    }
  },
  "auth": {
    "nextjsbloga585e058": {
      "AppClientID": "string",
      "AppClientIDWeb": "string",
      "CreatedSNSRole": "string",
      "IdentityPoolId": "string",
      "IdentityPoolName": "string",
      "UserPoolArn": "string",
      "UserPoolId": "string",
      "UserPoolName": "string"
    },
    "userPoolGroups": {
      "monopadusersGroupRole": "string"
    }
  },
  "function": {
    "S3Trigger71ba5706": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "UserPostConfirmation": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "crossresults": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "nextjsblog24b60079": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "omniGoResults": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "omnigoCategories": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "processFitFile": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "processGpxFile": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "publishPost": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "raceresult": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "raceresultgetResults": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "requestResultsProvider": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "routeAccessPoint": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "runSignupResults": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "runsignupCategories": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "shorturl": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "syncUserToSendgrid": {
      "Arn": "string",
      "CloudWatchEventRule": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "videoUploadToAsset": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "videoUploadUrl": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "videoWebhook": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "webscorer": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    },
    "webscorerResults": {
      "Arn": "string",
      "LambdaExecutionRole": "string",
      "LambdaExecutionRoleArn": "string",
      "Name": "string",
      "Region": "string"
    }
  },
  "storage": {
    "dynamo1a552de1": {
      "Arn": "string",
      "Name": "string",
      "PartitionKeyName": "string",
      "PartitionKeyType": "string",
      "Region": "string",
      "StreamArn": "string"
    },
    "routeFiles": {
      "BucketName": "string",
      "Region": "string"
    }
  }
}