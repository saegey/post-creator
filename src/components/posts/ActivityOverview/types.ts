export interface ActivityData {
  normalizedPower: number | null;
  elevationGain: number | null;
  distance: number | null;
  heartAnalysis: Record<string, number> | null;
  elapsedTime: {
    seconds: number;
  } | null;
  tempAnalysis: Record<string, number> | null;
  stoppedTime: number | null;
  powerAnalysis: Record<string, number> | null;
  cadenceAnalysis: Record<string, number> | null;
  // timeInRed: number | string | null;
}

export interface ActivityItem {
  title: string;
  value: string;
}

export type ActivityItems = Array<{ title: string; value: string }>;
