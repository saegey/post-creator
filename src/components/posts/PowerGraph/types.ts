// types.ts

export interface Coordinate {
  x: number;
  y: number;
}

export interface PowerCurveGraphProps {
  data: Coordinate[];
  ftp: number;
}
