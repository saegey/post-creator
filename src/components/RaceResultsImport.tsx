import {
  Flex,
  Box,
  Label,
  Input,
  Button,
  Text,
  Select,
  Spinner,
} from 'theme-ui';
import { API } from 'aws-amplify';
import React from 'react';

import { PostContext } from './PostContext';
import { EditorContext } from './EditorContext';
import RaceResultsPreview from './RaceResultsPreview';
import StandardModal from './StandardModal';
import BlackBox from './BlackBox';

const RaceResultsImport = ({ editor }) => {
  const [categories, setCategories] = React.useState<{
    data: { filterValues: Array<Array<string>> };
  }>();
  const [category, setCategory] = React.useState('');
  const [division, setDivision] = React.useState('');
  const [raceId, setRaceId] = React.useState('');
  const [key, setKey] = React.useState();
  const [server, setServer] = React.useState();
  const [previewResults, setPreviewResults] = React.useState(false);
  const [isLoading, setIsLoading] = React.useState(false);

  const { isRaceResultsModalOpen, setIsRaceResultsModalOpen } =
    React.useContext(EditorContext);

  const { setRaceResults } = React.useContext(PostContext);

  const getResults = async () => {
    if (!category || !division) {
      throw new Error('No category or division for results');
    }
    const url = `/raceresult/results?category=${encodeURIComponent(
      category
    )}&division=${encodeURIComponent(
      division
    )}&raceId=${raceId}&key=${key}&server=${server}`;
    const res = await API.get('api12660653', url, { response: true });

    const fields = res.data.list.Fields.map((f) => f.Label);

    const data = res.data.data.map((d) => {
      const temp = {};
      d.map((column, i) => {
        temp[fields[i - 1] ? fields[i - 1].replace(/\s+/g, '') : 'missing'] =
          column;
      });
      return temp;
    });

    return data;
  };

  const getCategories = async ({ url }) => {
    const resultsUrl = new URL(url);
    setRaceId(resultsUrl.pathname.split('/')[1]);

    if (resultsUrl.host === 'my.raceresult.com') {
      const apiName = 'api12660653';
      const path = `/raceresult?raceId=${resultsUrl.pathname.split('/')[1]}`;
      const myInit = {
        response: true, // OPTIONAL (return the entire Axios response object instead of only response.data)
      };
      const response = await API.get(apiName, path, myInit);
      // console.log(response);
      setKey(response.data.key);
      setServer(response.data.server);
      setCategories(response);
      return response;
    }
  };

  return (
    <>
      <StandardModal
        title={'Race Results'}
        setIsOpen={setIsRaceResultsModalOpen}
        isOpen={isRaceResultsModalOpen}
      >
        {previewResults && <RaceResultsPreview editor={editor} />}
        {!previewResults && (
          <Flex sx={{ gap: '10px', flexDirection: 'row' }}>
            <form
              onSubmit={(event: any) => {
                setIsLoading(true);
                event.preventDefault();
                const form = new FormData(event.target);
                const url = form.get('url') as string;
                getCategories({ url }).then(() => setIsLoading(false));
              }}
              style={{ width: '100%' }}
            >
              <Flex sx={{ gap: '20px', flexDirection: 'column' }}>
                <Box>
                  <Label htmlFor='url' variant='defaultLabel'>
                    Url
                  </Label>
                  <Input
                    id='url'
                    name='url'
                    variant={'defaultInput'}
                    readOnly={raceId ? true : false}
                  />
                </Box>
                {categories && (
                  <>
                    <Box>
                      <Label htmlFor='url' variant='defaultLabel'>
                        Category
                      </Label>
                      <Select
                        id='category'
                        variant={'defaultInput'}
                        onChange={(e) => {
                          setCategory(e.target.value);
                        }}
                      >
                        <option></option>
                        {categories.data?.filterValues[0].map(
                          (c: string, i: number) => (
                            <option key={`category-${i}`}>{c}</option>
                          )
                        )}
                      </Select>
                    </Box>
                    <Box>
                      <Label htmlFor='url' variant='defaultLabel'>
                        Division
                      </Label>
                      <Select
                        variant={'defaultInput'}
                        onChange={(e) => {
                          setDivision(e.target.value);
                        }}
                      >
                        <option></option>
                        {categories.data?.filterValues[1].map(
                          (c: string, i: number) => (
                            <option key={`category-${i}`}>{c}</option>
                          )
                        )}
                      </Select>
                    </Box>
                  </>
                )}
                {!categories && (
                  <Box sx={{ marginLeft: 'auto' }}>
                    <Button
                      disabled={isLoading ? true : false}
                      variant='primaryButton'
                    >
                      <Flex sx={{ gap: '10px' }}>
                        <Text as='span'>Import</Text>
                        {isLoading && (
                          <Spinner sx={{ size: '20px', color: 'white' }} />
                        )}
                      </Flex>
                    </Button>
                  </Box>
                )}
                {categories && (
                  <Box sx={{ marginLeft: 'auto' }}>
                    <Button
                      type='button'
                      variant='primaryButton'
                      onClick={() => {
                        setIsLoading(true);
                        getResults().then((r: any) => {
                          setRaceResults({
                            results: r as any,
                            selected: undefined,
                          });
                          setPreviewResults(true);
                          setIsLoading(false);
                        });
                      }}
                    >
                      <Flex sx={{ gap: '10px' }}>
                        <Text as='span'>Import</Text>
                        {isLoading && (
                          <Spinner sx={{ size: '20px', color: 'white' }} />
                        )}
                      </Flex>
                    </Button>
                  </Box>
                )}
              </Flex>
            </form>
          </Flex>
        )}
      </StandardModal>
    </>
  );
};

export default RaceResultsImport;
