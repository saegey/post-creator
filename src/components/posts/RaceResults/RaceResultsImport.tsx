import { Text } from "theme-ui";
import React from "react";

import { EditorContext } from "../Editor/EditorContext";
import RaceResultsPreview from "./RaceResults/RaceResultsPreview";
import WebscorerResultsPreview from "./WebScorer/WebscorerResultsPreview";
import StandardModal from "../../shared/StandardModal";
import CrossResultsPreview from "./CrossResults/CrossResultsPreview";
import OmniResultsPreview from "./OmniGo/OmniResultsPreview";
import RaceImportForm from "./RaceImportForm";
import {
  CrossResultsMetaType,
  OmniMetaType,
  RaceResultsMetaType,
  ResultsContext,
  RunSignupMetaType,
  WebScorerMetaType,
} from "./ResultsContext";
import RunSignUpResultsPreview from "./RunSignup/RunSignupResultsPreview";
import { useSlateContext } from "../../SlateContext";

const RaceResultsImport = () => {
  const [previewResults, setPreviewResults] = React.useState(false);
  const [previewWebscorerResults, setPreviewWebscorerResults] =
    React.useState(false);
  const [previewCrossResults, setPreviewCrossResults] = React.useState(false);
  const [previewOmniResults, setPreviewOmniResults] = React.useState(false);
  const [previewRunSignupResults, setPreviewRunSignupResults] =
    React.useState(false);
  const { editor } = useSlateContext();

  if (!editor) {
    return;
  }

  const [resultsUrl, setResultsUrl] = React.useState<string>("");

  const [raceResultsMeta, setRaceResultsMeta] =
    React.useState<RaceResultsMetaType>({
      category: "",
      categories: { data: { filterValues: [] } },
      division: "",
      key: "",
      server: "",
      eventName: "",
    });

  const [webScorerMeta, setWebScorerMeta] = React.useState<WebScorerMetaType>({
    category: "",
    categories: [],
    eventName: "",
  });

  const [crossResultsMeta, setCrossResultsMeta] =
    React.useState<CrossResultsMetaType>({
      category: "",
      categories: [],
      eventName: "",
    });

  const [omniMeta, setOmniMeta] = React.useState<OmniMetaType>({
    category: "",
    categories: [],
    eventName: "",
  });

  const [runSignupMeta, setRunSignupMeta] = React.useState<RunSignupMetaType>({
    category: undefined,
    categories: [],
    eventName: "",
    categoryName: "",
  });

  const { isRaceResultsModalOpen, setIsRaceResultsModalOpen, menuPosition } =
    React.useContext(EditorContext);
  const { path } = menuPosition;

  const isPreview = () => {
    return (
      previewResults ||
      previewWebscorerResults ||
      previewCrossResults ||
      previewOmniResults ||
      previewRunSignupResults
    );
  };

  const notPreview = () => {
    return (
      !previewResults &&
      !previewWebscorerResults &&
      !previewCrossResults &&
      !previewOmniResults &&
      !previewRunSignupResults
    );
  };

  return (
    <>
      <StandardModal
        title={"Race Results"}
        setIsOpen={setIsRaceResultsModalOpen}
        isOpen={isRaceResultsModalOpen}
      >
        <ResultsContext.Provider
          value={{
            previewResults,
            setPreviewResults,
            previewWebscorerResults,
            setPreviewWebscorerResults,
            previewCrossResults,
            setPreviewCrossResults,
            previewOmniResults,
            setPreviewOmniResults,
            resultsUrl,
            setResultsUrl,
            raceResultsMeta,
            setRaceResultsMeta,
            webScorerMeta,
            setWebScorerMeta,
            crossResultsMeta,
            setCrossResultsMeta,
            omniMeta,
            setOmniMeta,
            runSignupMeta,
            setRunSignupMeta,
            previewRunSignupResults,
            setPreviewRunSignupResults,
          }}
        >
          {isPreview() && (
            <Text as="div" sx={{ marginY: "5px" }}>
              Select your result below:
            </Text>
          )}

          {previewResults && <RaceResultsPreview path={path} />}
          {previewWebscorerResults && (
            <WebscorerResultsPreview editor={editor} path={path} />
          )}
          {previewCrossResults && <CrossResultsPreview editor={editor} />}
          {previewOmniResults && <OmniResultsPreview editor={editor} />}
          {previewRunSignupResults && (
            <RunSignUpResultsPreview editor={editor} />
          )}

          {notPreview() && <RaceImportForm />}
        </ResultsContext.Provider>
      </StandardModal>
    </>
  );
};

export default RaceResultsImport;
