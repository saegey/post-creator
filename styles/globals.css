@import url('https://fonts.googleapis.com/css2?family=Oxanium:wght@700&family=Public+Sans:ital,wght@0,200;0,400;0,600;0,700;1,400;1,500&display=swap');

* {
  margin: 0;
  padding: 0;
}

/* This allow me to have the full width of the page without the initial padding/margin*/
body,
html {
  margin: 0;
  padding: 0;
  /* height: 100%; */
  width: 100%;
  font-family: 'Public Sans', sans-serif;
  min-width: 700px;
  background-color: #f8f8f8;
}

.splitdiv {
  height: 100%;
  width: 50%;
}

/* This part contains all of the left side of the screen */
/* ----------------------------------------- */
#leftdiv {
  float: left;
  background-color: #fafafa;
  height: 932px;
}

#leftdivcard {
  margin: 0 auto;
  width: 50%;
  background-color: white;
  margin-top: 25vh;
  transform: translateY(-50%);
  box-shadow: 10px 10px 1px 0px rgba(78, 205, 196, 0.2);
  border-radius: 10px;
}

#leftbutton {
  background-color: #656565;
  border-radius: 5px;
  color: #fafafa;
  margin-left: 350px;
}

/* ----------------------------------------- */

/* This part contains all of the right side of the screen */
/* ----------------------------------------- */
#rightdiv {
  float: right;
  background-color: #cbcfcf;
  height: 932px;
}

#rightdivcard {
  margin: 0 auto;
  width: 50%;
  margin-top: 50vh;
  transform: translateY(-50%);
  background-position: bottom;
  background-size: 20px 2px;
  background-repeat: repeat-x;
}

/* ----------------------------------------- */

/* Basic styling */
/* ----------------------------------------- */

button {
  outline: none !important;
  font-family: 'Public Sans';
  margin-bottom: 15px;
  border: none;
  font-size: 20px;
  padding: 8px;
  padding-left: 20px;
  padding-right: 20px;
  margin-top: 0px;
  cursor: pointer;
  color: '#EEE';
}

h1 {
  font-family: 'Public Sans', sans-serif;
  /* text-align: center; */
  font-size: 40px;
}

input {
  font-family: 'Public Sans', sans-serif;
  font-size: 16px;
  font-family: 15px;
  /* border-color: #898989; */
}

input {
  width: 30%;
  /* height: 20px; */
  padding: 16px;
  margin-left: 1%;
  margin-right: 2%;
  margin-top: 15px;
  margin-bottom: 10px;
  display: inline-block;
  border: none;
  border-radius: 5px;
}


/* ----------------------------------------- */

.main {
  height: 100%;
  width: 100%;
  display: inline-block;
}

.main-h2 {
  padding-top: 20px;
  text-align: center;
}

.body-h1 {
  padding-top: 20px;
  text-align: center;
  color: white;
}

.inner-p {
  color: white;
  text-align: center;
}

.main-align {
  text-align: center;
}

.form-control {
  margin-left: 15px;
}

button:disabled,
button[disabled] {
  /* border: 1px solid #999999; */
  background-color: #cccccc;
  color: #666666;
}

@keyframes pulse-animation {
  0% {
    box-shadow: 0 0 0 0px rgba(0, 0, 0, 0.3);
  }

  100% {
    box-shadow: 0 0 0 10px rgba(0, 0, 0, 0);
  }
}

@keyframes shine-lines {
  0% {
    background-position: -100px;
  }

  40%,
  100% {
    background-position: 140px;
  }
}

@keyframes shine-avatar {
  0% {
    background-position: -32px;
  }

  40%,
  100% {
    background-position: 500px;
  }
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}
