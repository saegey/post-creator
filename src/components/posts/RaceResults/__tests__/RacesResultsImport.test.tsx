import { expect, jest, test } from "@jest/globals";

// const { expect, test, jest } = require("jest");
import RaceResultsImport from "../RaceResultsImport";
import renderer from "react-test-renderer";
import { CustomEditor } from "../../../../types/common";

describe("race results import", () => {
  test("it should render)", () => {
    const editor = jest.fn() as unknown as CustomEditor;
    const component = renderer.create(<RaceResultsImport editor={editor} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
