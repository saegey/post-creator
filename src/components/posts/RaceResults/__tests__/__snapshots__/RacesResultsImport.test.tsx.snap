// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`race results import it should render) 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1ynrwvv-Box{box-sizing:border-box;margin:0;min-width:0;position:fixed;top:0;height:100%;width:100%;left:0;background-color:rgba(var(--theme-ui-colors-blackBoxColor), 0.7);z-index:50;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
      }
    }
    data-emotion="css 1ynrwvv-Box"
  />,
  <div
    className="css-1ynrwvv-Box"
    onClick={[Function]}
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1clgg7s-Box{box-sizing:border-box;margin:0;min-width:0;max-width:690px;position:fixed;max-height:100%;height:100%;width:100%;margin:auto;background:background;padding:20px;}@media screen and (min-width: 40em){.css-1clgg7s-Box{position:inherit;max-height:800px;height:auto;border-radius:5px;}}@media screen and (min-width: 52em){.css-1clgg7s-Box{position:inherit;max-height:800px;height:auto;border-radius:5px;}}",
        }
      }
      data-emotion="css 1clgg7s-Box"
    />
    <div
      className="css-1clgg7s-Box"
      onClick={[Function]}
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-11a34ar-Flex{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom-width:1px;border-bottom-color:divider;border-bottom-style:solid;padding-top:5px;padding-bottom:5px;}",
          }
        }
        data-emotion="css 11a34ar-Flex"
      />
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1a3s0av-Flex{box-sizing:border-box;margin:0;min-width:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom-width:1px;border-bottom-color:divider;border-bottom-style:solid;padding-top:5px;padding-bottom:5px;}",
          }
        }
        data-emotion="css 1a3s0av-Flex"
      />
      <div
        className="css-1a3s0av-Flex"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-dz3wj2-Text{font-size:20px;font-weight:600;}",
            }
          }
          data-emotion="css dz3wj2-Text"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-r4uko2-Text{box-sizing:border-box;margin:0;min-width:0;font-size:20px;font-weight:600;}",
            }
          }
          data-emotion="css r4uko2-Text"
        />
        <div
          className="css-r4uko2-Text"
        >
          Race Results
        </div>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-afjjfe-Close{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:100%;margin-left:auto;}",
            }
          }
          data-emotion="css afjjfe-Close"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-wqkgnj-IconButton-Close{box-sizing:border-box;margin:0;min-width:0;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;padding:4px;width:32px;height:32px;color:inherit;background-color:transparent;border:none;border-radius:4px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:100%;margin-left:auto;}",
            }
          }
          data-emotion="css wqkgnj-IconButton-Close"
        />
        <button
          aria-label="Close"
          className="css-wqkgnj-IconButton-Close"
          id="close-button"
          onClick={[Function]}
          title="Close"
        >
          <svg
            fill="currentColor"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-9617yx-Flex{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:10px;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}",
          }
        }
        data-emotion="css 9617yx-Flex"
      />
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1wmeto1-Flex{box-sizing:border-box;margin:0;min-width:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:10px;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}",
          }
        }
        data-emotion="css 1wmeto1-Flex"
      />
      <div
        className="css-1wmeto1-Flex"
      >
        <form
          onSubmit={[Function]}
          style={
            {
              "width": "100%",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-civ7zb-Flex{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:20px;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
              }
            }
            data-emotion="css civ7zb-Flex"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-6n86f7-Flex{box-sizing:border-box;margin:0;min-width:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:20px;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
              }
            }
            data-emotion="css 6n86f7-Flex"
          />
          <div
            className="css-6n86f7-Flex"
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-1csqw02-Box{box-sizing:border-box;margin:0;min-width:0;margin-top:20px;}",
                }
              }
              data-emotion="css 1csqw02-Box"
            />
            <div
              className="css-1csqw02-Box"
            >
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-18qhrp0-Box{box-sizing:border-box;margin:0;min-width:0;width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
                  }
                }
                data-emotion="css 18qhrp0-Box"
              />
              <label
                className="css-18qhrp0-Box"
                htmlFor="url"
              >
                Link to results
              </label>
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-jhzmzz-Input{box-sizing:border-box;margin:0;min-width:0;display:block;width:100%;padding:8px;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;font-size:inherit;line-height:inherit;border:1px solid;border-radius:4px;color:inherit;background-color:transparent;}.css-jhzmzz-Input:autofill,.css-jhzmzz-Input:autofill:hover,.css-jhzmzz-Input:autofill:focus{box-shadow:inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);font-size:inherit;}.css-jhzmzz-Input:autofill:first-line,.css-jhzmzz-Input:autofill:hover:first-line,.css-jhzmzz-Input:autofill:focus:first-line{font-size:1rem;}.css-jhzmzz-Input:-webkit-autofill,.css-jhzmzz-Input:-webkit-autofill:hover,.css-jhzmzz-Input:-webkit-autofill:focus{box-shadow:inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);font-size:inherit;}.css-jhzmzz-Input:-webkit-autofill:first-line,.css-jhzmzz-Input:-webkit-autofill:hover:first-line,.css-jhzmzz-Input:-webkit-autofill:focus:first-line{font-size:1rem;}",
                  }
                }
                data-emotion="css jhzmzz-Input"
              />
              <input
                className="css-jhzmzz-Input"
                id="url"
                name="url"
                readOnly={false}
              />
            </div>
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-9rr9pc-Box{box-sizing:border-box;margin:0;min-width:0;margin-left:auto;}",
                }
              }
              data-emotion="css 9rr9pc-Box"
            />
            <div
              className="css-9rr9pc-Box"
            >
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-5unna8-Box{box-sizing:border-box;margin:0;min-width:0;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;display:inline-block;text-align:center;line-height:inherit;-webkit-text-decoration:none;text-decoration:none;font-size:inherit;padding-left:16px;padding-right:16px;padding-top:8px;padding-bottom:8px;color:white;background-color:primary;border:0;border-radius:4px;}",
                  }
                }
                data-emotion="css 5unna8-Box"
              />
              <button
                className="css-5unna8-Box"
                disabled={false}
                id="import-results"
              >
                <style
                  dangerouslySetInnerHTML={
                    {
                      "__html": ".css-donn1z-Flex{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:10px;}",
                    }
                  }
                  data-emotion="css donn1z-Flex"
                />
                <style
                  dangerouslySetInnerHTML={
                    {
                      "__html": ".css-fdyw5a-Flex{box-sizing:border-box;margin:0;min-width:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:10px;}",
                    }
                  }
                  data-emotion="css fdyw5a-Flex"
                />
                <div
                  className="css-fdyw5a-Flex"
                >
                  <style
                    dangerouslySetInnerHTML={
                      {
                        "__html": ".css-g89m8z-Box{box-sizing:border-box;margin:0;min-width:0;}",
                      }
                    }
                    data-emotion="css g89m8z-Box"
                  />
                  <span
                    className="css-g89m8z-Box"
                  >
                    Import
                  </span>
                </div>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
]
`;
